### PR TITLE
Update operatorkit to v4.2.0 

### PR DESCRIPTION
**Pull request by mig4**
Towards: https://github.com/giantswarm/giantswarm/issues/15588

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [ ] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: service/controller/managementcluster/controller.go [hunk 1]
```diff
@@ -78,7 +70,9 @@ func NewController(config ControllerConfig) (*Controller, error) {
 				return new(v1.Service)
 			},
 			Resources: resources,
-			Selector:  selector,
+			Selector: labels.SelectorFromSet(labels.Set{
+				"component": "apiserver",
+			}),
```
Comment (mig4) [inline service/controller/managementcluster/controller.go:75]: This is the only change apart from import path changes - `controller.Selector` is no longer a thing and we can just use a labels selector directly.

**Comment by mig4** [comment]
Note: like the others, this targets the branch in #324, it's split out for easier review, the tests will be fixed once all required changes are combined.

New graph after updating `operatorkit` (starting to look nicer 😁 ):

prometheus-meta-operator gomod-graph 4-after-operatorkit

**Review by QuentinBisson — APPROVED** [review]

File: service/controller/managementcluster/controller.go (hunk 1 above)
Comment (TheoBrigitte) [reply service/controller/managementcluster/controller.go:75]: :+1:

**Review by TheoBrigitte — APPROVED** [review]
Looks valid to me, but make sure CI is green

**Pull request merged**